core/server/services/auth/api-key/admin.js

```diff
@@ -90,10 +91,12 @@ const authenticate = (req, res, next) => {
         // https://github.com/auth0/node-jsonwebtoken/issues/208#issuecomment-231861138
         const secret = Buffer.from(apiKey.get('secret'), 'hex');
 
-        // @TODO When v3 api hits we should check against the api actually being used
+        const {pathname} = url.parse(req.originalUrl);
+        const [hasMatch, version = 'v2', api = 'admin', resource = ''] = pathname.match(/ghost\/api\/([^/]+)\/([^/]+)\/(.+)*/); // eslint-disable-line no-unused-vars
+
         // ensure the token was meant for this api
         const options = Object.assign({
-            audience: '/v2/admin/'
+            audience: new RegExp(`\/?${version}\/${api}(?:\/${resource})?\/?$`) // eslint-disable-line no-useless-escape
```

Comment:
Think we need to be clear in the commit message why the `resource` bit is being added and link to some reference at least?

Reply:
Hmm, I think its probably just better to remove it, there is no clear use-case for it at all at the moment and doesn't pin us in a corner either if not done now, so keeping it to api version is better and we can extend it later when needed. Wdyt?

Reply:
the less code - the better :fire:

Reply:
🔥 done